data/src/main/kotlin/com/vultisig/wallet/data/api/swapAggregators/SwapKitApi.kt

```
@@ -0,0 +1,130 @@
+                if (!response.status.isSuccess()) {
+                    val body = runCatching { response.bodyAsText() }.getOrNull().orEmpty()
+                    val code = extractErrorCode(body)
+                    val status = HttpStatusCode.fromValue(response.status.value).description
+                    throw SwapKitError.fromCode(
+                        code = code,
+                        fallbackMessage = body.ifEmpty { status },
+                    )
+                }
+                try {
+                    response.body<T>()
+                } catch (e: SerializationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)